### PR TITLE
[JUJU-136] enable juju_machine_lock to display unit machine lock status.

### PR DESCRIPTION
**Pull request by hmlanigan**
Add back the introspection worker for units to show unit machine lock info in `juju_machine_lock` command in the output.  This was lost when the unit and machine agents were combined into one process.

This update should allow for application agent machine lock status as well.

Includes some refactoring and clean up.  Move the addons package to a different location, to avoid pulling pieces of cmd into worker.  Nor does the introspection worker startup required an entire agent, it only needs the agent tag.

## QA steps

Bootstrap, deploy ubuntu and ntp charms, then relate them. 

```console
$ juju ssh ubuntu/0
$  juju_machine_lock
machine-0:
  holder: none
unit-ntp-0:
  holder: ntp/0 uniter (run install hook), holding 42s
unit-ubuntu-0:
  holder: none
```

### CODE REVIEW

File: worker/introspection/script.go [hunk 1]
```diff
@@ -114,7 +114,9 @@ juju_statetracker_report () {
 }
 
 juju_machine_lock () {
-  juju_agent machinelock 2> /dev/null
+  for agent in $(ls /var/lib/juju/agents); do
```
Comment (wallyworld) [inline worker/introspection/script.go:117]: Are these sorted by default? QA steps seem to say yes?

**Review by wallyworld — APPROVED** [review]
Awesome, thanks for fixing

File: worker/introspection/script.go (hunk 1 above)
Comment (hmlanigan) [reply worker/introspection/script.go:117]: `ls` does sort by default.

**Comment by hmlanigan** [comment]
$$merge$$

**Pull request merged**